### PR TITLE
Convert Dashboard API endpoints to use ClickHouse

### PR DESCRIPTION
**Pull request by roborourke**
Fixes https://github.com/humanmade/altis-accelerate/issues/70

### CODE REVIEW

File: inc/api/namespace.php [hunk 1]
```diff
@@ -1024,81 +809,84 @@ function get_post_diff_data( array $post_ids, $start, $end, $resolution = '1d' )
...
-			$res = Utils\query( $query, [
-				'preference' => __FUNCTION__,
-			] );
+	// Generate fields for histogram of current date range.
+	$buckets = [];
+	$bucket_start = $start;
+	while ( $bucket_start < $end ) {
+		$bucket_end = strtotime( "+{$resolution}", $bucket_start );
+		$buckets[ $bucket_start ] = sprintf(
+			'uniqCombined64If(endpoint_id, event_timestamp >= toDateTime64(%1$d,3) AND event_timestamp < toDateTime64(%2$d,3)) as `%1$d`',
```
Comment (shadyvb) [inline inc/api/namespace.php:838]: This looks a bit weird, particularly the last `as` bit:
```
as `%1$d`
```

**Review by shadyvb — APPROVED** [review]
LGTM, with one little observation which might be okay.

File: inc/api/namespace.php [hunk 2]
```diff
@@ -223,285 +205,146 @@ function get_query( array $events, int $start, int $end, array $aggs, ?Filter $f
...
+			count() as views,
+			-- uniques
+			uniqCombined64(endpoint_id) as uniques,
+			-- returning
+			uniqCombined64If(endpoint_id, endpoint_metrics['sessions'] > 1) as returning,
+			-- bounce
+			uniqCombined64If(endpoint_id, endpoint_metrics['pageViews'] < 2) as bounce
+		FROM analytics
+		WHERE
+			attributes['blogId'] = %s
```
Comment (rmccue) [inline inc/api/namespace.php:232]: Does Clickhouse support parameterised queries? That'd be better than building a raw query

Comment (roborourke) [reply inc/api/namespace.php:232]: Appears so, wasn't aware of that feature https://clickhouse.com/docs/en/interfaces/http/#cli-queries-with-parameters

File: inc/api/namespace.php (hunk 1 above)
Comment (roborourke) [reply inc/api/namespace.php:838]: I'm naming the field selected as the timestamp for the start of the histogram bucket, makes the post-processing very minimal.

File: inc/api/namespace.php (hunk 2 above)
Comment (roborourke) [reply inc/api/namespace.php:232]: @rmccue I'll do a follow up to parameterise the queries.

**Pull request merged**